File: editor/en/[205]Isomorphic Strings.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> map1 = new HashMap<>();
        Map<Character, Character> map2 = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char sc = s.charAt(i);
            char tc = t.charAt(i);
            Character mappedToT = map1.get(sc);
            Character mappedToS = map2.get(tc);

            if (mappedToT == null) {
                map1.put(sc, tc);
            } else if (!mappedToT.equals(tc)) {
                return false;
            }

            if (mappedToS == null) {
                map2.put(tc, sc);
            } else if (!mappedToS.equals(sc)) {
                return false;
            }
        }

        return true;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[205]Isomorphic Strings.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        map1 = {}
        map2 = {}

        for char_s, char_t in zip(s, t):
            if map1.get(char_s, char_t) != char_t or map2.get(char_t, char_s) != char_s:
                return False
            map1[char_s] = char_t
            map2[char_t] = char_s

        return True

# leetcode submit region end(Prohibit modification and deletion)
